### PR TITLE
#1335: NumerEnvelope wrap function restored

### PR DESCRIPTION
**Pull request by ruca905**
#1335

### CODE REVIEW

File: src/main/java/org/cactoos/scalar/NumberEnvelope.java [hunk 1]
```diff
@@ -41,78 +41,34 @@
...
-        return new Unchecked<>(this.fnum).value();
+    public long longValue() {
+        return origin.longValue();
     }
 
     @Override
-    public final double doubleValue() {
-        return new Unchecked<>(this.dnum).value();
+    public float floatValue() {
+        return origin.floatValue();
```
Comment (andreoss) [inline src/main/java/org/cactoos/scalar/NumberEnvelope.java:67]: @ruca905 Please qualify field access with `this.`

File: src/main/java/org/cactoos/scalar/NumberOf.java [hunk 2]
```diff
@@ -65,49 +65,26 @@ public NumberOf(final String txt) {
         this(new TextOf(txt));
     }
 
+    public NumberOf(final Scalar<Long> lnm, final Scalar<Integer> inm,
```
Comment (andreoss) [inline src/main/java/org/cactoos/scalar/NumberOf.java:68]: @ruca905 Could you place each argument on separate line?

File: src/main/java/org/cactoos/scalar/MaxOf.java [hunk 3]
```diff
@@ -235,7 +235,7 @@ public MaxOf(final Float... src) {
                     }
                 }
                 return max;
-            });
+            }));
```
Comment (andreoss) [inline src/main/java/org/cactoos/scalar/MaxOf.java:238]: @ruca905 It would be match better not to cluster parenthesis this way

**Review by andreoss — CHANGES_REQUESTED** [review]

**Comment by andreoss** [comment]
@ruca905 The built is failing, I belive you have many qulice complains as well

**Comment by andreoss** [comment]
@ruca905 Any update on this?

File: src/main/java/org/cactoos/scalar/NumberEnvelope.java [hunk 4]
```diff
@@ -41,78 +41,34 @@
...
-        this.fnum = fnm;
-        this.dnum = dnm;
+    public NumberEnvelope(final Number number) {
+        this.origin = number;
     }
 
     @Override
-    public final int intValue() {
-        return new Unchecked<>(this.inum).value();
+    public Double value() throws Exception {
```
Comment (victornoel) [inline src/main/java/org/cactoos/scalar/NumberEnvelope.java:51]: @ruca905 please remove the implementation of `Scalar<Double>`, it's out of scope of `NumberEnvelope`.

File: src/main/java/org/cactoos/scalar/NumberEnvelope.java [hunk 5]
```diff
@@ -41,78 +41,34 @@
...
-     */
-    public NumberEnvelope(final Scalar<Long> lnm, final Scalar<Integer> inm,
-        final Scalar<Float> fnm, final Scalar<Double> dnm) {
-        super();
-        this.lnum = lnm;
-        this.inum = inm;
-        this.fnum = fnm;
-        this.dnum = dnm;
+    public NumberEnvelope(final Number number) {
+        this.origin = number;
```
Comment (victornoel) [inline src/main/java/org/cactoos/scalar/NumberEnvelope.java:47]: @ruca905 please also delegate byteValue, shortValue, toString, equals and hashCode

File: src/main/java/org/cactoos/scalar/NumberOf.java [hunk 6]
```diff
@@ -65,49 +60,28 @@ public NumberOf(final String txt) {
         this(new TextOf(txt));
     }
 
+    public NumberOf(final Scalar<Long> lnm,
+                    final Scalar<Integer> inm,
+                    final Scalar<Float> fnm,
+                    final Scalar<Double> dnm) {
+        this(dnm);
```
Comment (victornoel) [inline src/main/java/org/cactoos/scalar/NumberOf.java:67]: @ruca905 I think there was some kind of confusion about what `NumberOf` and `Sealed` were about… so, now that it's clearer to me, I think we should merge `Sealed` and `NumberOf` into `NumberOf`, for this:
- we move the old code based on multiple scalars from `NumberEnvelope` to `NumberOf`
- we move the constructors of `Sealed` to `NumberOf`
- we keep the text/string based constructor here and delegate to the scalar-based ones
- we delete `Sealed`

File: src/test/java/org/cactoos/io/GzipInputTest.java [hunk 7]
```diff
@@ -67,7 +63,7 @@ public void readFromGzipInput() throws Exception {
         ).affirm();
     }
 
-    @Test(expected = EOFException.class)
+    @Test(expected = UncheckedIOException.class)
```
Comment (victornoel) [inline src/test/java/org/cactoos/io/GzipInputTest.java:66]: @ruca905 please revert all changes from this file

File: src/test/java/org/cactoos/io/GzipOutputTest.java [hunk 8]
```diff
@@ -86,7 +81,7 @@ public void writeToGzipOutput() throws Exception {
         ).affirm();
     }
 
-    @Test(expected = IOException.class)
+    @Test(expected = UncheckedIOException.class)
```
Comment (victornoel) [inline src/test/java/org/cactoos/io/GzipOutputTest.java:84]: @ruca905 please revert all changes from this file

**Review by victornoel — CHANGES_REQUESTED** [review]
@ruca905 here are some last changes to finish this PR. Ping me when it's done :)

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @andreoss[/z](https://www.0crat.com/u/andreoss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (66 days), architects (@victornoel) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Pull request [#1488](https://github.com/yegor256/cactoos/issues/1488) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by sereshqua** [comment]
@ruca905 please make sure that all problems addressed by code reviewers would be fixed by you in next PRs, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request closed without merging**